packages/@dcl/inspector/src/lib/data-layer/host/utils/undo-redo-array.ts

```diff
@@ -0,0 +1,66 @@
+import { UndoRedo } from '../undo-redo'
+
+const ONE_MB_IN_BYTES = 1_048_576
+const ONE_GB_IN_BYTES = ONE_MB_IN_BYTES * 1024
+
+export function UndoRedoArray(maxEntries: number, maxSize: number = ONE_GB_IN_BYTES) {
+  const valueList: UndoRedo[] = []
+  let memorySize: number = 0
+
+  function valueSize(val?: UndoRedo) {
+    let fileSize: number = 0
+    if (val?.$case === 'file' && val) {
+      for (const opreation of val.operations) {
+        fileSize += opreation.newValue?.length ?? 0
+        fileSize += opreation.prevValue?.length ?? 0
+      }
+    }
+    return fileSize
+  }
+
+  return {
+    push(value: UndoRedo) {
+      // If we reach the max entries, remove the first element and calc size
+      while (valueList.length >= maxEntries) {
+        const valToRemove = valueList.shift()
+        memorySize -= valueSize(valToRemove)
+      }
+
+      // Add current value size to the memory
+      const currentSize = valueSize(value)
+      memorySize += currentSize
+
+      // add the value to the list.
+      valueList.push(value)
+
+      // if we reach the max_size limit, start removing files.
+      while (memorySize >= maxSize) {
+        for (const index in valueList) {
```

Comment:
`for in` doesn't ensure return order for arrays, I think we should use just a `for` loop (or `forEach`, etc)

Reply:
and you will be able to avoid casting `index` to a number :P

Reply:
mmm are u sure about the order? I think that was for some browsers in the past.
I'll change anyways but if this is the case then we have to refactor in a lot of places.